IMG_avif.c

```diff
@@ -185,7 +185,7 @@ typedef struct
     SDL_RWops *src;
     Uint64 start;
     uint8_t *data;
-    Sint64 size;
+    Uint64 size;
```

Comment:
This breaks `out->size <= 0` check

Reply:
I suggest the following (to be applied after reverting this one + hopefully didn't miss anything)
```diff
diff --git a/IMG_avif.c b/IMG_avif.c
index 4f51282..e587330 100644
--- a/IMG_avif.c
+++ b/IMG_avif.c
@@ -199,13 +199,13 @@ static avifResult ReadAVIFIO(struct avifIO * io, uint32_t readFlags, uint64_t of
         return AVIF_RESULT_IO_ERROR;
     }
 
-    if (size > context->size) {
+    if (size > (Uint64)context->size) {
         uint8_t *data = (uint8_t *)SDL_realloc(context->data, size);
         if (!data) {
             return AVIF_RESULT_IO_ERROR;
         }
         context->data = data;
-        context->size = size;
+        context->size = (Sint64)size;
     }
 
     out->data = context->data;
```